src/app/update-user-dialog/update-user-dialog.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit, Inject } from "@angular/core";
+import { MatDialogRef, MAT_DIALOG_DATA } from "@angular/material";
+import { UserService } from "../user.service";
+import { User } from "../user";
+
+@Component({
+  selector: "app-update-user-dialog",
+  templateUrl: "./update-user-dialog.component.html",
+  styleUrls: ["./update-user-dialog.component.scss"]
+})
+export class UpdateUserDialogComponent implements OnInit {
+  user: User;
+
+  constructor(
+    private userService: UserService,
+    @Inject(MAT_DIALOG_DATA) data
+  ) {
+    this.user = { ...data.user };
+  }
+
+  ngOnInit() {}
+
+  confirm(user: User) {
+    if (false) {
```

Comment:
you received comment in previous PRs related to such (smelly constructions) and here it is.

Reply:
sorry, done